src/simulation/LoadGenerator.cpp

```diff
@@ -1819,24 +1990,27 @@ LoadGenerator::waitTillComplete(GeneratedLoadConfig cfg)
     if (inconsistencies.empty() && sorobanInconsistencies.empty() &&
         cfg.isDone())
     {
-        CLOG_INFO(LoadGen, "Load generation complete.");
-        mLoadgenComplete.Mark();
-        reset();
+        // Check whether run met the minimum success rate for soroban invoke
+        if (checkMinimumSorobanSuccess(cfg))
+        {
+            CLOG_INFO(LoadGen, "Load generation complete.");
+            mLoadgenComplete.Mark();
+            reset();
+        }
+        else
+        {
+            CLOG_INFO(LoadGen, "Load generation failed to meet minimum success "
+                               "rate for soroban invoke transactions.");
+            emitFailure(false);
```

Comment:
Do we need to reset soroban state here as well?

Reply:
No, the persistent soroban state is valid in this case and does not need resetting (there are no inconsistencies). It's safe for the user to continue generating load without re-invoking one of the soroban setup modes. If the user does want to reset soroban state, then re-invoking one of the soroban setup modes will accomplish that. I added a comment to make that more clear in 6647f00a58dc3aa3f46ec35ed342a1c94bc0544f.